docs/usage.md

```diff
@@ -40,3 +40,37 @@ Note that Ruff's pre-commit hook should run before Black, isort, and other forma
 
 Ruff can also be used as a [VS Code extension](https://github.com/charliermarsh/ruff-vscode) or
 alongside any other editor through the [Ruff LSP](https://github.com/charliermarsh/ruff-lsp).
+
+Ruff can also be used as a [GitHub Action](https://github.com/features/actions) :
+
+You can use Ruff within a GitHub Actions workflow without setting your own Python environment. Great for enforcing your Ruff rules.  Naturally, assumes the repo has an appropriate [config](https://beta.ruff.rs/docs/configuration/) in a `pyproject.toml`, `ruff.toml`, or `.ruff.toml`.
+
+Compatibility
+This action is known to support all GitHub-hosted runner OSes. In addition, only published versions of Ruff are supported (i.e. whatever is available on PyPI).
+
+Usage
+Create a file named .github/workflows/ruff.yml inside your repository with:
+
+```yaml
+name: Ruff
+
+on: [push, pull_request]
+
+jobs:
+  ruff:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: charliermarsh/ruff@v0
```

Comment:
How does the versioning work? Where is the current version expressed?

Reply:
https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-tags-for-release-management

Ultimately, I'd likely either use v0 for now [ until ruff is v1 ] where action will then keep with ruff versions, or can pin it as tied to release tags via existing process.  Version can also be specified [ version of ruff, which can be different from the version of action ... I assume almost always these will be the same ] in the with: options.

Reply:
Does that make sense?

Reply:
Okay thanks, I might just need to spend a few mins reading through these docs before merging to make sure I understand what's going on. But I'll try not to bother you further :)

Reply:
Happy to help, if I can be useful.  If you wanted to talk through, even could hop on a quick call for pointers [ email at austin@apache.org to set that up -- I'll also remove this comment with that address shortly ], I'm open the next little bit or could schedule something further out.  

Makes sense.  If you merge, you need to know you [ and/or the community with write permissions ] is prepared to release and support.

Reply:
Probably the thing to really be aware of is:

* The action can be released as part of tag/release process [ that you already have ].  
* The action version can differ from the ruff version used by the action, if it is specified.  

And, I added you (@charliermarsh ) as collaborator in https://github.com/brucearctor/testing_repo/actions ... to be able to see logs of prior runs, so you can have some more info on how this worked.  [ that repo was running actions based on tags/releases from https://github.com/brucearctor/ruff/ ex: https://github.com/brucearctor/ruff/tree/v0.0.009  ]

Reply:
Oh okay, so would folks need to use (e.g.) `v0.0.260` (our next release) or later?

Reply:
Well, they would need to use an action based on a tag [ once merged, an `action.yml` file -- along with associated code for the composite action -- will exist in the repo, and at some point you'll wind up tagging a specific point with a version number ].  So, the action called would first be available with that tag.  

Someone could use the action based on that tag with a prior version of ruff ->

```
name: Ruff

on: [push, pull_request]

jobs:
  ruff:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: charliermarsh/ruff@v0.0.260
        with:
          version: v0.0.249
          options: --select ANN
```

^^ Would be the action of version 0.0.260, which would run ruff version 0.0.249.  

By default, the action always installs/runs on the most recent version of ruff.  Using the with: version: is the way to allow users to pin a specific version [ ex: I keep the same version pinned in pre-commit, and our codebases ].

Reply:
Yeah makes sense. So the _action_ version will be available starting in the first tag that's created after this is merged. But they can specify any Ruff version within the action, as above. (I assume Black works similarly?)

Reply:
> Yeah makes sense. So the _action_ version will be available starting in the first tag that's created after this is merged. But they can specify any Ruff version within the action, as above. (I assume Black works similarly?)

That is correct.


Black:  I *think* yes.  But, I would need to check to be 100%.  Personally, I trust black enough not to have bothered with minor versions [ I just rely on `uses: psf/black@stable` ].  It is my understanding, based on ruff SemVer, that we aren't thinking about a v1 release yet, and therefore somewhat early to be relying on a stable path.  The pace of ruff development makes me think pinning versions at this point is prudent [ esp. in an action, so the development in ruff doesn't introduce a new rule that gets added which then fails the pipeline, even though not related to code change ].  Black is different, in that it is doing something different.

Reply:
Added a little more to the docs

Reply:
Thanks for expanding the documentation. I prefer versioning the action independently of ruff because users may get notified by dependentbot that their action is outdated and needs upgrading, even though nothing has changed. It's also unclear if I could use the ruff action 0.0.259 to install ruff 0.0.258 or 0.0.260. 

What I've seen other repositories doing is that they don't have a `version` option but instead use the action tag `@0.0.259` to determine the ruff version to install. I'm undecided if that would be a good idea (dependentbot strikes again). 

We currently promote using a package manager like `pip` to install ruff because it solves the versioning problem; All users use the `ruff` version configured for the project. What's your opinion on whether we should fallback to the ruff version specified in the repository rather than the latest ruff version?

Reply:
I also prefer versioning the action independently, but would we then need to release it as its own repo to avoid coupling it to our own release tags?

Reply:
@MichaReiser : Does dependabot actually highlight if using not the latest action?  I have not experienced that [ but I also do not specify minor versions of actions.  yes, specify minor versions of the version of code that they call certainly  ]

Ex, the following is what someone would place into something like `.github/workflows/ruff.yml` which just specifies to use the most recent v0 release with the latest version of ruff  -->

```yaml
name: Ruff
on: [push, pull_request]
jobs:
  ruff:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: charliermarsh/ruff@v0
``` 

Ex, the following is what someone would place into something like `.github/workflows/ruff.yml` which just specifies to use the most recent v0 release along with the 0.0.260 version of ruff -->

```yaml
name: Ruff
on: [push, pull_request]
jobs:
  ruff:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: charliermarsh/ruff@v0
        with:
            version: v0.0.260
``` 


I do not know whether I have ever seen minor versions of actions suggested to be actually used as part of workflows.

Reply:
To me the concern over versioning of the action itself is overkill, also, would highlight - since comparisons were made - black doesn't look to have this problem.  It just includes as same repo/releases.